pkg/distro/arch/Dockerfile.tmpl

```diff
@@ -0,0 +1,43 @@
+# syntax = docker.io/docker/dockerfile:1.4@sha256:9ba7531bd80fb0a858632727cf7a112fbfd19b17e94c4e84ced81e24ef1a0dbc
+# Generated by repro-get.
+
+# Dockerfile for building a container image using the hash file.
+
+# ⚠️  EXPERIMENTAL ⚠️
+
+# Usage:
+# Make sure that the hash file "SHA256SUMS-{{.OCIArchDashVariant}}" is present in the current directory.
+# ----------------------------------------------------------
+# export DOCKER_BUILDKIT=1
+# docker build .
+# ----------------------------------------------------------
+
+ARG BASE_IMAGE={{.BaseImage}} # {{.BaseImageOrig}}
+ARG REPRO_GET_PROVIDER={{join .Providers ","}}
+
+ARG REPRO_GET_VERSION={{.ReproGetVersion}}
+ARG REPRO_GET_SHA256SUMS_SHA256SUM={{.ReproGetSHASHA}}
+{{snippet "fetch-repro-get"}}
+
+FROM --platform=${TARGETPLATFORM} ${BASE_IMAGE}
+ARG TARGETARCH
+ARG TARGETVARIANT
+ARG REPRO_GET_PROVIDER
+SHELL ["/bin/bash", "-c"]
+# The cache dir is mounted under a directory inside tmpfs (/dev/*), so that the mount point directory does not remain in the image
+RUN \
+  --mount=type=cache,target=/dev/.cache/repro-get \
+  --mount=type=bind,from=repro-get,source=/repro-get.linux-${TARGETARCH}${TARGETVARIANT:+-${TARGETVARIANT}},target=/usr/local/bin/repro-get \
+  --mount=type=bind,source=.,target=/mnt \
+    set -eux -o pipefail ; \
+    export SOURCE_DATE_EPOCH="$(stat --format=%Y /etc/pacman.conf)" && \
+    pacman-key --init && \
+    /usr/local/bin/repro-get --provider="${REPRO_GET_PROVIDER}" --cache=/dev/.cache/repro-get install "/mnt/SHA256SUMS-${TARGETARCH}${TARGETVARIANT:+-${TARGETVARIANT}}" && \
+    : Remove unneeded files for reproducibility && \
+    rm -rf /etc/pacman.d/gnupg && \
```

Comment:
?

Reply:
`pacman-key` is a thin wrapper around gnupg, and instead of "trust every key in our keyring package" it evaluates the pgp web-of-trust client side during an upgrade to ensure every packager key is trusted by multiple Arch Linux master keys (as a defense in depth mechanism, Debian and Alpine are more pragmatic in that regard).

To process this graph, gnupg needs to use `lsign` and to do _that_ it needs access to a private key (that's what `pacman-key --init` does, among other things). This key should not remain in the final image tho, leaking the private key while the public key is also still trusted in the final image would be _very bad_, for that reason it's best to just nuke the whole folder that contains the pgp web-of-trust stuff after we're done with it.

Reply:
Thanks, could you add that explanation to the code comment lines?

Reply:
Done